app/tests/fixtures/open-secret/fixture.ts

```diff
@@ -0,0 +1,186 @@
+   */
+  headers?: { [key: string]: string };
+
+  /**
+   * JSON response.
+   */
+  responseData: T;
+
+  /**
+   * Response status code, defaults to `200`.
+   */
+  status?: number;
+
+  /**
+   * How often a mock should be used. By default, it will be used every time.
+   */
+  times?: number;
+};
+
+type OpenSecretApiMock = {
+  /**
+   * Sets up open secret api mock with raw response data.
+   */
+  setup: <T = unknown>(options: OpenSecretApiMockOptions<T>) => Promise<void>;
+  /**
+   * Sets up open secret api mock with encrypted response data.
+   */
+  setupEncrypted: <T = unknown>(
+    options: OpenSecretApiMockOptions<T>,
+  ) => Promise<void>;
+};
+
+export type OpenSecretFixture = {
+  /**
+   * Open secret API mock. Use to mock the Open Secret responses.
+   */
+  openSecretApiMock: OpenSecretApiMock;
+  /**
+   * Presets Open Secret session before starting a test.
+   * True by default. Set to false if you want to test "session handshake"
+   */
+  presetOpenSecretSession?: boolean;
+};
+
+type MockData = {
+  config: OpenSecretApiMockOptions<unknown>;
+  numOfInvocations: number;
+};
+
+const openSecretBaseUrl = 'https://preview-enclave.opensecret.cloud';
+
+export const test = base.extend<OpenSecretFixture>({
+  presetOpenSecretSession: [true, { option: true }],
+  openSecretApiMock: async (
+    { page, baseURL, presetOpenSecretSession },
+    use,
+  ) => {
+    const mocksMap = new Map<string, MockData>();
+    let openSecretRequestsCounter = 0;
+
+    page.on('request', (request) => {
+      if (request.url().startsWith(openSecretBaseUrl)) {
+        openSecretRequestsCounter++;
+      }
+    });
+
+    if (presetOpenSecretSession) {
+      await page.addInitScript(
+        async ({ baseURL, session }) => {
+          if (
+            window.location.origin === baseURL &&
+            !window.sessionStorage.getItem('sessionId') &&
+            !window.sessionStorage.getItem('sessionKey')
+          ) {
+            window.sessionStorage.setItem('sessionId', session.id);
+            window.sessionStorage.setItem('sessionKey', session.key);
+          }
+        },
+        { baseURL, session },
+      );
+    }
+
+    const setup: OpenSecretApiMock['setup'] = async (options) => {
+      const optionsStr = JSON.stringify(options);
+      const key = crypto.createHash('sha256').update(optionsStr).digest('hex');
+      const mockData = mocksMap.get(key);
+      if (!mockData) {
+        mocksMap.set(key, { config: options, numOfInvocations: 0 });
+      }
+
+      const { url, headers, responseData, status, times } = options;
+
+      await page.route(
+        getFullUrl(url),
+        async (route) => {
+          await route.fulfill({
+            status,
+            headers: headers,
+            json: responseData,
+          });
+          const mockData = mocksMap.get(key);
+          if (!mockData) {
+            throw new Error('Mock data not found');
```

Comment:
This should never happen, right? Above, this `key` always gets set.

Reply:
correct. if it is here the mock data has to be set. this is just to satisfy the ts